lib/pos_blockchain.go

```diff
@@ -262,15 +262,44 @@ func (bc *Blockchain) validateBlockHeight(desoBlock *MsgDeSoBlock) error {
 // validateBlockView validates the view for a given block. First, it checks that
 // the view is greater than the latest committed block view. If not,
 // we return an error indicating that we'll never accept this block. Next,
-// it checks that the view is less than or equal to the latest uncommitted block.
+// it checks that the view is less than or equal to its parent.
 // If not, we return an error indicating that we'll want to add this block as an
 // orphan. Then it will check if that the view is exactly one greater than the
 // latest uncommitted block if we have an regular vote QC. If this block has a
 // timeout QC, it will check that the view is at least greater than the latest
 // uncommitted block's view + 1.
 func (bc *Blockchain) validateBlockView(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
+	committedBlock, _ := bc.getHighestCommittedBlock()
+	if committedBlock == nil {
+		// This is an edge case we'll never hit in practice since all the PoW blocks
+		// are committed.
+		return errors.New("validateBlockView: No committed blocks found")
+	}
+	// Validate that the view is greater than the latest committed block view.
+	if desoBlock.Header.ProposedInView <= committedBlock.Header.ProposedInView {
+		return RuleErrorPoSBlockViewEarlierThanCommittedBlock
+	}
```

Comment:
What's the intended purpose of comparing the block's view against the committed tip's? It's a bit redundant since we'll later on need to validate the block's view against the node's current view, which is a cheaper validation.

Reply:
Nixed this validation.